core/common/src/main/java/alluxio/exception/AlluxioException.java

```
@@ -65,13 +59,16 @@ public AlluxioTException toAlluxioTException() {
    * @param e the {link AlluxioTException} to convert to a {@link AlluxioException}
    * @return a {@link AlluxioException} of the type specified in e, with the message specified in e
    */
+  @SuppressWarnings("unchecked")
   public static AlluxioException from(AlluxioTException e) {
-    AlluxioExceptionType exceptionType = AlluxioExceptionType.valueOf(e.getType());
-    Class<? extends AlluxioException> throwClass = exceptionType.getExceptionClass();
+    Class<? extends AlluxioException> throwClass;
     try {
-      return throwClass.getConstructor(String.class).newInstance(e.getMessage());
+      throwClass = (Class<? extends AlluxioException>) Class.forName(e.getType());
+      AlluxioException throwInstance =
```

why create the variable throwInstance instead of returning directly?